edx_django_utils/db/read_replica.py

```diff
@@ -0,0 +1,134 @@
+"""
+Tools for making queries read from the read-replica database, rather than from the writer database.
+
+Settings:
+    EDX_READ_REPLICA_DB_NAME: The name of the read-replica in the DATABASES django setting.
+        Defaults to "read_replica".
+    EDX_WRITER_DB_NAME: The name of the writer in the DATABASES django setting.
+        Defaults to "writer".
+"""
+from contextlib import contextmanager
+from functools import partial
+import threading
+
+from django.conf import settings
+
+
+READ_REPLICA_NAME = getattr(settings, "EDX_READ_REPLICA_DB_NAME", "read_replica")
+WRITER_NAME = getattr(settings, "EDX_WRITER_DB_NAME", "default")
+
+READ_REPLICA_OR_DEFAULT = (
+    READ_REPLICA_NAME if READ_REPLICA_NAME in settings.DATABASES else WRITER_NAME
+)
+
+
+def use_read_replica_if_available(queryset):
+    """
+    If there is a database called 'read_replica',
+    use that database for the queryset / manager.
+
+    Example usage:
+        queryset = use_read_replica_if_available(SomeModel.objects.filter(...))
+
+    Arguments:
+        queryset (QuerySet)
+
+    Returns: QuerySet
+    """
+    return queryset.using(READ_REPLICA_OR_DEFAULT)
+
+
+def read_replica_or_default():
+    """
+    If there is a database called READ_REPLICA_DB,
+    return READ_REPLICA_DB, otherwise return WRITER_NAME.
+
+    This function is similiar to `use_read_replica_if_available`,
+    but is be more syntactically convenient for method call chaining.
+    Also, it always falls back to WRITER_NAME,
+    no matter what the queryset was using before.
+
+    Example usage:
+        queryset = SomeModel.objects.filter(...).using(read_replica_or_default())
+
+    Returns: str
+    """
+    return READ_REPLICA_OR_DEFAULT
+
+
+@contextmanager
+def read_queries_only(db_name):
+    """
+    A context manager that sets all reads inside it to be from the read-replica.
+    """
+    old_db_name = _storage.db_name
+    _storage.db_name = READ_REPLICA_NAME
+    try:
+        yield
+    finally:
+        _storage.db_name = old_db_name
+
+
+@contextmanager
+def write_queries(db_name):
+    """
+    A context manager that sets all reads inside it to be from the writer.
```

Comment:
The name might be fine.  Could you provide more context on when someone would want to use this?  That may indicate a name change, but maybe not.